src/unix/linux_like/android/mod.rs

```diff
@@ -517,6 +517,11 @@ s! {
         pub ifr6_prefixlen: u32,
         pub ifr6_ifindex: c_int,
     }
+
+    // is __be16 __bitwise __u16
+    pub struct __c_anonymous___be16 {
```

Comment:
Consider deriving Eq/PartialEq, adding a method to access/set it as host `u16`, and adding a `Debug` based on the host representation?

If you do that, your later stuff on `ethhdr` can just be a derive.